libsolidity/ast/Types.cpp

```diff
@@ -165,11 +165,33 @@ string identifierList(TypePointer const& _type1, TypePointer const& _type2)
 
 string parenthesizeUserIdentifier(string const& _internal)
 {
-	return parenthesizeIdentifier(boost::algorithm::replace_all_copy(_internal, "$", "$$$"));
+	return parenthesizeIdentifier(_internal);
 }
 
 }
 
+string Type::escapeIdentifier(string const& _identifier)
+{
+	string ret = _identifier;
+	// FIXME: should be _$$$_
```

Comment:
Why not do that?

Reply:
I didn't wanted to change the currently identifiers, they were using `$` -> `$$$`